eng/ci/templates/official/jobs/run-benchmarks.yml

```
@@ -0,0 +1,121 @@
+parameters:
+- name: functionAppName
+  type: string
+- name: description
+  type: string
+- name: storeResultsInDatabase
+  type: boolean
+  default: false
+- name: additionalCrankArgs
+  type: string
+  default: ''
+
+jobs:
+- job: ${{ parameters.functionAppName }}
+
+  variables:
+    runDescription: ${{ parameters.description }}
+    functionApp: ${{ parameters.functionAppName }}
+    functionAppOutputPath: $(Build.ArtifactStagingDirectory)\FunctionApps\$(functionApp)
+    benchmarkResultsJsonPath: "$(Build.ArtifactStagingDirectory)\\BenchmarkResults\\$(functionApp).json"
+    functionsWorkerRuntime: 'dotnet-isolated'
+    crankAgentUrl: "http://localhost:5010"  # Default crank agent URL.
+
+  templateContext:
+    outputParentDirectory: $(Build.ArtifactStagingDirectory)
+    outputs:
+    - output: pipelineArtifact
+      displayName: Publish benchmark results
+      path: $(benchmarkResultsJsonPath)
+      artifact: 'BenchmarkResults_$(functionApp)'
+
+  steps:
+
+  - checkout: self
+    path: sourcecode/host
+
+  - task: AzureKeyVault@2
+    condition: and(succeeded(), eq('${{ parameters.storeResultsInDatabase }}', true))
+    inputs:
+      azureSubscription:  Azure-Functions-Host-CI-internal
+      KeyVaultName: functions-perf-crank-kv
+      SecretsFilter: BenchmarkResultsSqlConnectionString
+      RunAsPreJob: false
+
+  - task: PowerShell@2
+    condition: and(succeeded(), eq('${{ parameters.storeResultsInDatabase }}', true))
+    displayName: Set environment variables
+    inputs:
+      targetType: 'inline'
+      script: |
+        $env:CRANK_SQL_CONNECTION_STRING = "$(BenchmarkResultsSqlConnectionString)"
+        Write-Host "##vso[task.setvariable variable=CRANK_SQL_CONNECTION_STRING]$env:CRANK_SQL_CONNECTION_STRING"
```

can be simplified to, and remove line 51
```suggestion
        Write-Host "##vso[task.setvariable variable=CRANK_SQL_CONNECTION_STRING]$(BenchmarkResultsSqlConnectionString)"
```

But I see later on this is passed in manually, so is the env variable rename even needed? `BenchmarkResultsSqlConnectionString` will already be an environment variable accessible throughout this pipeline. All variables are env vars for all tasks.

---

Removed the env variable and passing the connection string value directly now.